datalad/core/local/run.py

```diff
@@ -623,7 +624,12 @@ def format_command(dset, command, **kwds):
     command = normalize_command(command)
     sfmt = SequenceFormatter()
 
-    for k, v in dset.config.items("datalad.run.substitutions"):
+    for k in set(cfg_defs.keys()).union(dset.config.keys()):
+        v = dset.config.get(
```

Comment:
explicit better than implicit, so we do not have some surprises if something overlaps in the names subspace after we do `k.replace` below, could we have smth like:

```suggestion
        if not k.startswith("datalad.run.substitutions."):
            continue
        v = dset.config.get(
```
?

Reply:
Makes a lot of sense. Feel free to commit, when all issues are expressed. I want to avoid the expensive CI reruns.

Reply:
I took another look while updating the -next patch. I went for this https://github.com/datalad/datalad-next/pull/488

Reply:
shouldn't we have it here?

Reply:
I don't think it matters now. This is a technical detail that could be rectified at the end, if desired. You asked much broader questions to which I had no answers. I don't know if and in which direction this PR could move forward. Once that is clear to me, I will take another look.

I have released the patch with datalad-next 1.0.1, so the immediate needs are addressed and this PR can move in its own pace.

Reply:
in effect that next patch to do the same, so let's try this suggested one finally here now.